lab_3_generate_by_ngrams/main.py

```diff
@@ -222,6 +326,16 @@ def generate_next_token(self, sequence: tuple[int, ...]) -> Optional[dict]:
         In case of corrupt input arguments, None is returned
         """
 
+        if not (isinstance(sequence, tuple) and sequence
+                and len(sequence) >= self._n_gram_size - 1):
+            return None
+        next_tokens = {}
+        context = sequence[- self._n_gram_size + 1:]
+        for n_gram, freq in self._n_gram_frequencies.items():
+            if n_gram[: self._n_gram_size - 1] == context:
```

Comment:
There is a simpler way to index the last element, same for the next line

Reply:
Fixed